src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1782,6 +1784,25 @@ private IDeclarationPatternOperation CreateBoundDeclarationPatternOperation(Boun
             return new DeclarationPattern(variable, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
+        private IDiscardPatternOperation CreateBoundDiscardPatternOperation(BoundDiscardPattern boundDiscardPattern)
+        {
+            SyntaxNode syntax = boundDiscardPattern.Syntax;
+            ITypeSymbol type = null;
+            Optional<object> constantValue = default(Optional<object>);
+            bool isImplicit = boundDiscardPattern.WasCompilerGenerated;
+            return new DiscardPattern(_semanticModel, syntax, type, constantValue, isImplicit);
+        }
+
+        private IRecursivePatternOperation CreateBoundRecursivePatternOperation(BoundRecursivePattern boundRecursivePattern)
+        {
+            ISymbol variable = boundRecursivePattern.Variable;
+            SyntaxNode syntax = boundRecursivePattern.Syntax;
+            ITypeSymbol type = null;
+            Optional<object> constantValue = default(Optional<object>);
+            bool isImplicit = boundRecursivePattern.WasCompilerGenerated;
+            return new RecursivePattern(variable, _semanticModel, syntax, type, constantValue, isImplicit);
```

Comment:
>return new RecursivePattern(variable, _semanticModel, syntax, type, constantValue, isImplicit); [](start = 12, length = 95)

All constituents of BoundRecursivePattern originating from syntax should be exposed. Otherwise, they will be undiscoverable from the IOperation tree, which will be the source of bugs.

Reply:
That is an open issue in [https://github.com/dotnet/roslyn/issues/23222](https://github.com/dotnet/roslyn/issues/23222)

---
In reply to: [156537223](https://github.com/dotnet/roslyn/pull/23209#discussion_r156537223) [](ancestors = 156537223)